Mage.Tests/src/test/java/org/mage/test/cards/single/woc/BlightwingBanditTest.java

```diff
@@ -0,0 +1,47 @@
+package org.mage.test.cards.single.woc;
+
+import mage.constants.PhaseStep;
+import mage.constants.Zone;
+import org.junit.Test;
+import org.mage.test.serverside.base.CardTestPlayerBase;
+
+/**
+ * @author Susucr
+ */
+public class BlightwingBanditTest extends CardTestPlayerBase {
+
+    /**
+     * {@link mage.cards.b.BlightwingBandit Blightwing Bandit} {3}{B}
+     * Creature — Faerie Rogue
+     * Flying, deathtouch
+     * Whenever you cast your first spell during each opponent’s turn, look at the top card of that player’s library, then exile it face down. You may play that card for as long as it remains exiled, and mana of any type can be spent to cast it.
+     * 2/2
+     */
+    private static final String bandit = "Blightwing Bandit";
+
+    @Test
+    public void test_Simple() {
+        setStrictChooseMode(true);
+        skipInitShuffling();
+
+        addCard(Zone.BATTLEFIELD, playerA, bandit);
+        addCard(Zone.BATTLEFIELD, playerA, "Mountain", 3);
+        addCard(Zone.HAND, playerA, "Lightning Bolt");
+        addCard(Zone.LIBRARY, playerB, "Fire // Ice", 1); // split card just to check it works on both faces
+
+        castSpell(2, PhaseStep.UPKEEP, playerA, "Lightning Bolt", playerB);
+        waitStackResolved(2, PhaseStep.UPKEEP, 2); // resolve trigger from bandit
+        checkPlayableAbility("can cast fire", 2, PhaseStep.UPKEEP, playerA, "Cast Fire", true);
+        checkPlayableAbility("can cast ice", 2, PhaseStep.UPKEEP, playerA, "Cast Ice", true);
+        castSpell(2, PhaseStep.UPKEEP, playerA, "Ice", bandit);
```

Comment:
Strange commands here.

You are using `waitStackResolved(2, PhaseStep.UPKEEP, 2)` to keep something in stack? If it's a full resolve then must use simple params without 2.

If you want to keep in stack something and cast Ice then `castSpell(2, PhaseStep.UPKEEP, playerA, "Ice", bandit)` is wrong cause it will be cast after full stack resolve. You must use castSpell with spellOnStack param to execute it with stack objects.

Reply:
More details about diff between check/run/show and cast/play commands for a stack logic: https://github.com/magefree/mage/pull/8583#issuecomment-1046104765

Reply:
You're right, I didn't mean to keep the spell in stack, but wanted to wait for the trigger to resolve for Fire // Ice to get exiled.